### PR TITLE
ci: skip optimize from platform release

### PR DESCRIPTION
**Pull request by OmranAbazid**
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Comment by OmranAbazid** [comment]
Not sure if we should also add it on the `mvnw -B versions:set-property` in the workflow file. 
I did not do that but let me know if this is necessary.

**Review by nicpuppa — APPROVED** [review]

**Review by buccarel — APPROVED** [review]

File: .github/workflows/camunda-platform-release.yml [hunk 1]
```diff
@@ -155,6 +155,7 @@ jobs:
             -DpushChanges=${PUSH_CHANGES} \
             -DremoteTagging=${PUSH_CHANGES} \
             -DcompletionGoals="spotless:apply" \
+            -D skipOptimize \
```
Comment (buccarel) [inline .github/workflows/camunda-platform-release.yml:158]: I might be wrong, but I think the recommended syntax is without space, i.e.,
```
-DskipOptimize
```
I think in this case it's equivalent though

Comment (nicpuppa) [reply .github/workflows/camunda-platform-release.yml:158]: AFAIK it can be a problem on windows OS. Should be good in our case

**Pull request merged**